src/emulator/cpu.ts

```diff
@@ -370,7 +396,7 @@ export class Cpu {
 
             case Operation.ret: {
                 const condition = this.evaluateCondition(instruction);
-                const cycles = instruction.cycles + (instruction.opcode !== 0xc9 ? (condition ? 1 : -2) : 0);
+                const cycles = instruction.cycles + ((instruction.opcode !== 0xc9 && condition) ? 3 : 0);
```

Comment:
Let's set *all* incarnations of `RET` to 3 cycles in the definition below (including the unconditional one) and just add 3 more cycles here if the branch is taken.